community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/idmapping/string/EncodingIdMapper.java

```diff
@@ -414,8 +415,14 @@ private int detectAndMarkCollisions( ProgressListener progress )
             progress.add( batch );
         }
         progress.done();
-        monitor.numberOfCollisions( numberOfCollisions );
-        return numberOfCollisions;
+
+        if ( numberOfCollisions > Integer.MAX_VALUE )
```

Comment:
If someone overflows a long, we'll all be retired or dead by the time they come to report the problem.